src/gpio.rs

```diff
@@ -190,6 +190,7 @@ mod marker {
 impl<MODE> marker::Interruptable for Output<MODE> {}
 impl marker::Interruptable for Input {}
 impl marker::Readable for Input {}
+impl<const A: u8, MODE> marker::Readable for Alternate<A, MODE> {}
```

Comment:
It's a bit... Odd to read an AF pin IMO, but I referenced the datasheet and it's certainly allowed on this architecture. Ref section 11.3.12 of the RM